ibis_substrait/compiler/translate.py

```diff
@@ -968,16 +968,19 @@ def _extractdatefield(
     compiler: SubstraitCompiler,
     **kwargs: Any,
 ) -> stalg.Expression:
+    # e.g. "ExtractYear" -> "YEAR"
+    span = type(op).__name__.lstrip("Extract").upper()
```

Comment:
I've always found `{l,r}strip` to have surprising behavior:

```
>>> 'ExtractYear'.lstrip(''.join(reversed('tcartxE')))
'Year'
>>> 'ExtractYear'.lstrip('tcartxE')
'Year'
```

Another way might be `type(op).__name__[len("Extract"):]`

Not a big deal though.